x-pack/legacy/plugins/fleet/public/pages/agent_list/index.tsx

```diff
@@ -0,0 +1,222 @@
+        defaultMessage: 'Actions',
+      }),
+      actions: [
+        {
+          render: () => {
+            return (
+              <EuiLink color="primary" onClick={() => {}}>
+                <FormattedMessage
+                  id="xpack.fleet.agentList.viewActionLinkText"
+                  defaultMessage="view"
+                />
+              </EuiLink>
+            );
+          },
+        },
+      ],
+      width: '100px',
+    },
+  ];
+
+  const sorting = {
+    sort: {
+      field: 'last_checkin',
+      direction: 'asc',
+    },
+  };
+
+  const pagination = {
+    initialPageSize: 20,
+    pageSizeOptions: [10, 20, 50],
+  };
+
+  const search = {
+    box: {
+      incremental: true,
+      schema: true,
+    },
+    filters: [
+      {
+        type: 'field_value_selection',
+        field: 'policy_id',
+        name: i18n.translate('xpack.fleet.agentList.policyFilterLabel', {
+          defaultMessage: 'Policy',
+        }),
+        multiSelect: true,
+        options: [...new Set(agents.map(agent => agent.policy_id))].map(policy => ({
+          value: policy,
+        })),
+      },
+    ],
+    toolsRight: (
+      <EuiFlexGroup gutterSize="m" justifyContent="spaceAround">
+        <EuiFlexItem>
+          <EuiButton color="secondary" iconType="refresh" onClick={fetchAgents}>
+            <FormattedMessage id="xpack.fleet.agentList.reloadButton" defaultMessage="Reload" />
+          </EuiButton>
+        </EuiFlexItem>
+        <EuiFlexItem>
+          <EuiButton fill iconType="plusInCircle">
+            <FormattedMessage
+              id="xpack.fleet.agentList.addButton"
+              defaultMessage="Install new agent"
+            />
+          </EuiButton>
+        </EuiFlexItem>
+      </EuiFlexGroup>
+    ),
+  };
+
+  const emptyPrompt = (
+    <EuiEmptyPrompt
+      title={
+        <h2>
+          <FormattedMessage
+            id="xpack.fleet.agentList.noAgentsPrompt"
+            defaultMessage="No agents installed"
+          />
+        </h2>
+      }
+      actions={
+        <EuiButton fill iconType="plusInCircle">
+          <FormattedMessage
+            id="xpack.fleet.agentList.addButton"
+            defaultMessage="Install new agent"
+          />
+        </EuiButton>
+      }
+    />
+  );
+
+  return (
+    <EuiPageBody>
+      <EuiPageContent>
+        <EuiTitle size="l">
+          <h1>
+            <FormattedMessage id="xpack.fleet.agentList.pageTitle" defaultMessage="Elastic Fleet" />
+          </h1>
+        </EuiTitle>
+        <EuiSpacer size="s" />
+        <EuiTitle size="s">
+          <EuiText color="subdued">
+            <FormattedMessage
+              id="xpack.fleet.agentList.pageDescription"
+              defaultMessage="Use agents to faciliate data collection for your Elastic stack."
+            />
+          </EuiText>
+        </EuiTitle>
+        <EuiSpacer size="m" />
+        <EuiInMemoryTable
```

Comment:
this component has built in pagination but will likely run into problems once there are thousands of items. my intention is to refactor this into EuiBasicTable with manual pagination after this week since I see that we intend to support pagination in the API endpoints via `page` query param (correct me if I'm wrong!)